### ISSUE TITLE
Using latest version of ml-agents

### ISSUE THREAD

**Issue by maystroh** [mbaske/ml-agents-frame-stacking#1]
Thanks for sharing this work.  I integrated your work in my project using `ml-agents` v0.5 and it works like a charm. Now I need to use the latest version of `ml-agents` where the brain is not a scriptable object anymore. Do you have any idea where to start?

Thanks again for your effort.

**Comment by mbaske** [comment]
Hi @maystroh - With the latest ml-agents, it's probably better to first do any frame-stacking operations on your visual input, and then feed the result to the brain's custom render texture fields. Version 0.5 brains didn't have those, which is why my script had to intercept the camera registered to a brain and write frame-stacking results back to `brainParameters.cameraResolutions`.

**Comment by maystroh** [comment]
Thanks for your reply. I will try to do it and post the project here once done.

**Comment by maystroh** [comment]
@mbaske , what do you mean by "on your visual input" ? in the camera monobehavior class you mean?

**Comment by mbaske** [comment]
Yes, you can set up your camera to render into a render texture, and then do frame-stacking operations using that texture. https://docs.unity3d.com/ScriptReference/RenderTexture.html

**Comment by maystroh** [comment]
I see.. but how can I use visualObservations.cs class for instance in the new setup?

**Comment by mbaske** [comment]
I'm afraid there is no quick fix to get the code working with the current version. You'd need to change it so that it reads and writes from/to your own textures, rather than what's being provided by the agent brain. Ideally, the frame-stacking code would be independent of the context - that is wether it's being used with ml-agents or elsewhere. I haven't worked on such a generalized solution yet, because I suspect it would be more efficient to implement that with shaders (which is somewhat beyond my skills right now), as opposed to C#.
